tools/windows-node-installer/pkg/cloudprovider/aws/ec2.go

```diff
@@ -672,9 +741,13 @@ func (a *AwsProvider) findWindowsWorkerSg(infraID string) (string, error) {
 		},
 	})
 	if err != nil || len(sgs.SecurityGroups) == 0 {
-		return "", fmt.Errorf("worker security group not found, %v", err)
+		return nil, fmt.Errorf("worker security group not found, %v", err)
 	}
-	return *sgs.SecurityGroups[0].GroupId, nil
+
+	// In the case that we found the security group, get all rules we need for user's IP
+	_, err = a.verifyAndUpdateSg(sgs.SecurityGroups[0], vpc)
```

Comment:
`findSecurityGroup()` should only find the SG. `verfiyAndUpdateSg` should be called elsewhere.